### PR TITLE
feat: implement browse viewmodel and integrate into browse fragment

### PR DESCRIPTION
**Pull request by kdvuong**

### CODE REVIEW

File: app/src/main/java/com/cmput301f20t21/bookfriends/entities/Request.java [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
+    // empty constructor to be use by Document.toObject()
+    public Request() {
+    }
+
+    public Request(String id, String requester, String bookId, REQUEST_STATUS status) {
         this.id = id;
         this.requester = requester;
         this.bookId = bookId;
-        this.status = REQUEST_STATUS.valueOf(status);
+        this.status = status;
```
Comment (kdvuong) [inline app/src/main/java/com/cmput301f20t21/bookfriends/entities/Request.java:21]: I'm not sure how status will be converted when using `Document.toObject()`

Comment (c25vdw) [reply app/src/main/java/com/cmput301f20t21/bookfriends/entities/Request.java:21]: it works. screenshot of some logging in the `getAllRequestsByUsername` method
image

**Review by c25vdw — APPROVED** [review]
lgtm

**Review by Meilin-Lyu — APPROVED** [review]
Lgtm

**Pull request merged**